tests/components/lcn/test_device_trigger.py

```diff
@@ -55,25 +51,22 @@ async def test_get_triggers_module_device(hass, entry):
     assert_lists_same(triggers, expected_triggers)
 
 
-@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
-async def test_get_triggers_non_module_device(hass, entry):
+async def test_get_triggers_non_module_device(hass, lcn_connection):
     """Test we get the expected triggers from a LCN non-module device."""
     not_included_types = ("transmitter", "transponder", "fingerprint", "send_keys")
 
-    await init_integration(hass, entry)
     device_registry = dr.async_get(hass)
     for device_id in device_registry.devices:
         device = device_registry.async_get(device_id)
-        if device.model.startswith(("LCN host", "LCN group", "LCN resource")):
+        identifier = next(iter(device.identifiers))
+        if (identifier[1].count("-") != 1) or device.model.startswith("LCN group"):
```

Comment:
Can we turn the approach around and iterate over the known device identifiers and get the devices for these identifiers so we don't need to include conditional logic in the test?

Reply:
Done, Hope I got you right.